mypy/stats.py

```diff
@@ -152,12 +161,17 @@ def visit_unary_expr(self, o: UnaryExpr) -> None:
     def process_node(self, node: Expression) -> None:
         if self.all_nodes:
             typ = self.typemap.get(node)
-            if typ:
-                self.line = node.line
-                self.type(typ)
-
-    def type(self, t: Type) -> None:
-        if isinstance(t, AnyType):
+            self.line = node.line
+            self.type(typ)
+
+    def type(self, t: Optional[Type]) -> None:
+        if not t:
+            # if an expression does not have a type, it is often due to dead code
+            # we should not keep track of the number of these we encounter because there can be
+            # an unanalyzed value on a line with other analyzed expressions
+            self.record_line(self.line, TYPE_UNANALYZED)
+            return
```

Comment:
I don't think that would work - there are two separate if-statements and I think it's easier to just address it in the beginning and do an early return.